### PR TITLE
Added named query for every ManyToOne relation.

### PR DESCRIPTION
**Pull request by TheSprtCz**
# Overview
* Added named query for every ManyToOne relation.
* Refactored many of the one letter variables.
* Added Creator field into Topic.
* Added ThesisStart field into TopicApplication.
* Updated migrations.

### CODE REVIEW

**Comment by sbunciak** [comment]
Please use imperative for commit messages. See other useful tips here https://chris.beams.io/posts/git-commit/

File: studenthub-portal/src/main/java/cz/studenthub/core/TopicApplication.java [hunk 1]
```diff
@@ -54,6 +58,7 @@
   private TopicDegree degree;
 
   private Date thesisFinish;
+  private Date thesisStart;
```
Comment (sbunciak) [inline studenthub-portal/src/main/java/cz/studenthub/core/TopicApplication.java]: Please extract this change to a separate commit.

File: studenthub-portal/src/main/java/cz/studenthub/core/User.java [hunk 2]
```diff
@@ -43,8 +43,11 @@
 
 @Entity
 @Table(name = "Users")
-@NamedQueries({ @NamedQuery(name = "User.findAll", query = "SELECT u FROM User u"),
-    @NamedQuery(name = "User.findByRole", query = "SELECT u FROM User u join u.roles role WHERE role = :role") })
+@NamedQueries({ @NamedQuery(name = "User.findAll", query = "SELECT user FROM User user"),
+    @NamedQuery(name = "User.findByRole", query = "SELECT user FROM User user join user.roles role WHERE role = :role"),
+    @NamedQuery(name = "User.findByCompany", query = "SELECT user FROM User user WHERE user.company = :company"),
+    @NamedQuery(name = "User.findByRoleAndFaculty", query = "SELECT user FROM User user join user.roles role WHERE user.faculty = :faculty and role = :role"),
```
Comment (sbunciak) [inline studenthub-portal/src/main/java/cz/studenthub/core/User.java]: Why introducing this named query? It's either unnecessary or we are missing User.findByRoleAndCompany also.

File: studenthub-portal/src/main/java/cz/studenthub/core/Topic.java [hunk 3]
```diff
@@ -59,6 +64,9 @@
   @ManyToOne
   private User techLeader;
 
+  @ManyToOne
+  private User creator;
```
Comment (sbunciak) [inline studenthub-portal/src/main/java/cz/studenthub/core/Topic.java]: I don't see a reason for this relation. We already have a person who "owns" the topic from a technical POV. Perhaps need to be renamed but I'm sure we don't need a new field for it.

File: studenthub-portal/src/main/resources/db/changelog/db.changelog-1.0.xml [hunk 4]
```diff
@@ -27,14 +27,14 @@
...
-    <changeSet author="sbunciak (generated)" id="1485077755684-3" objectQuotingStrategy="QUOTE_ALL_OBJECTS">
+    <changeSet author="phala (generated)" id="1489497998265-3" objectQuotingStrategy="QUOTE_ALL_OBJECTS">
         <createTable catalogName="STUDENT-HUB" schemaName="PUBLIC" tableName="TOPICAPPLICATIONS">
             <column autoIncrement="true" name="ID" type="BIGINT(19)">
                 <constraints primaryKey="true" primaryKeyName="CONSTRAINT_B"/>
             </column>
             <column name="DEGREE" type="VARCHAR(255)"/>
             <column name="GRADE" type="VARCHAR(255)"/>
-            <column name="OFFICIALASSIGNMENT" type="LONGVARCHAR"/>
+            <column name="OFFICIALASSIGNMENT" type="VARCHAR(2147483647)"/>
```
Comment (sbunciak) [inline studenthub-portal/src/main/resources/db/changelog/db.changelog-1.0.xml]: I'm assuming this change is because of PgSQL is not supporting VARCHAR?

File: studenthub-portal/src/main/resources/db/changelog/db.changelog-1.0.xml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.1" encoding="UTF-8" standalone="no"?>
 <databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
-    <changeSet author="sbunciak (generated)" id="1485077755684-1" objectQuotingStrategy="QUOTE_ALL_OBJECTS">
+    <changeSet author="phala (generated)" id="1489497998265-1" objectQuotingStrategy="QUOTE_ALL_OBJECTS">
```
Comment (sbunciak) [inline studenthub-portal/src/main/resources/db/changelog/db.changelog-1.0.xml]: Do we really need to change the author here?

File: studenthub-portal/src/main/resources/db/changelog/db.changelog-1.0.xml [hunk 6]
```diff
@@ -43,22 +43,26 @@
...
                 <constraints primaryKey="true" primaryKeyName="CONSTRAINT_9"/>
             </column>
-            <column name="DESCRIPTION" type="LONGVARCHAR"/>
+            <column name="DESCRIPTION" type="VARCHAR(2147483647)"/>
             <column name="SHORTABSTRACT" type="VARCHAR(255)"/>
             <column name="TITLE" type="VARCHAR(255)">
                 <constraints nullable="false"/>
             </column>
             <column name="TECHLEADER_ID" type="BIGINT(19)"/>
+            <column name="CREATOR_ID" type="BIGINT(19)">
```
Comment (sbunciak) [inline studenthub-portal/src/main/resources/db/changelog/db.changelog-1.0.xml]: This is to be removed.

File: studenthub-portal/src/main/resources/db/changelog/db.changelog-1.0.xml (hunk 5 above)
Comment (TheSprtCz) [inline studenthub-portal/src/main/resources/db/changelog/db.changelog-1.0.xml]: I didn't do that on purpose, it was done by Liquibase itself, but I can change it back.

File: studenthub-portal/src/main/java/cz/studenthub/core/User.java (hunk 2 above)
Comment (TheSprtCz) [inline studenthub-portal/src/main/java/cz/studenthub/core/User.java]: In endpoints structure there was only companies/leaders, I assumed that only leader will have specified company so there was no need to create **findByRoleAndCompany**. **findByRoleAndFaculty** is there because we need to separate students and supervisors and I didn't want to create two namedQueries.

Comment (sbunciak) [inline studenthub-portal/src/main/java/cz/studenthub/core/User.java]: Tech.leader and Company Rep roles should have a company associated.

File: studenthub-portal/src/main/java/cz/studenthub/core/TopicApplication.java (hunk 1 above)
Comment (TheSprtCz) [inline studenthub-portal/src/main/java/cz/studenthub/core/TopicApplication.java]: Done.

File: studenthub-portal/src/main/java/cz/studenthub/core/User.java (hunk 2 above)
Comment (TheSprtCz) [inline studenthub-portal/src/main/java/cz/studenthub/core/User.java]: Added User.findByRoleAndCompany.

File: studenthub-portal/src/main/java/cz/studenthub/core/Topic.java (hunk 3 above)
Comment (TheSprtCz) [inline studenthub-portal/src/main/java/cz/studenthub/core/Topic.java]: Renamed techLeader to creator.
